src/prompt/term.rs

```diff
@@ -0,0 +1,125 @@
+pub(crate) struct TermSize {
+    window_max_rows: usize,
+    window_pos: usize,
+}
+
+impl Default for TermSize {
+    fn default() -> Self {
+        let mut window_max_rows = usize::MAX;
+
+        if let Some(termsize) = get_term_size() {
+            window_max_rows = (termsize.rows as usize)
+                .checked_sub(3)
+                .unwrap_or(termsize.rows as usize);
+        }
```

Comment:
I realized that it just not needed. Try to shrink your terminal to a few rows of height, and then run the `cargo run --example filter`:  the list is gonna be forever of a few rows, even if you expand the terminal height.

So, with `max_rows = usize::MAX` by default, the lists are gonna be working in an old way, which is good for backward compatibility.